app/main/views/g_cloud.py

```diff
@@ -544,6 +544,59 @@ def end_search(framework_framework, project_id):
     )
 
 
+@direct_award.route(
+    '/<string:framework_framework>/projects/<int:project_id>/did-you-award-contract',
+    methods=['GET', 'POST']
+)
+def did_you_award_contract(framework_framework, project_id):
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(all_frameworks, framework_framework)
+
+    # Get the requested Direct Award Project.
+    project = data_api_client.get_direct_award_project(project_id=project_id)['project']
+
+    form = DidYouAwardAContractForm()
+
+    if form.validate_on_submit():
```

Comment:
This should have an additional check of the request method

```
    if request.method == "POST" and form.validate_on_submit():
```

Reply:
The `validate_on_submit` method is from Flask-WTF, it includes the check for a `POST` already https://flask-wtf.readthedocs.io/en/stable/api.html#flask_wtf.FlaskForm.validate_on_submit